### DISCUSSION TITLE
Make a better usage example

### DISCUSSION THREAD

**Discussion by adamsd5** [Blazored/LocalStorage#179] [Ideas]
I'm new.  The usage example is not a typical use case.  I can't think of any reason you would write a thing to local storage and then immediately read it.  A better example is to show how to check if a value is stored locally first and use it if it does.  And if it doesn't, initialize it.   How do I do that?  Even better, how would I make a property that persists to local storage?  What does that code look like?

(The example I'm referring to:
        await localStorage.SetItemAsync("name", "John Smith");
        var name = await localStorage.GetItemAsync<string>("name");
)

**Comment by chrissainty** [comment]
Hi @adamsd5, I'm always happy to improve the samples.  Do you have any ideas about specific examples we could use?

**Comment by MaddoScientisto** [comment]
The way I always do it is as such:

Assuming we want to retrieve a Settings object (but could be anything)
 ```csharp
if (await _localStorage.ContainKeyAsync("Settings") == false)
{
    return new Settings(); // Init this however you like
}

return await _localStorage.GetItemAsync<Settings>("Settings");
```
